pkgs/build-support/ocaml/dune.nix

```diff
@@ -2,7 +2,8 @@
 
 { pname, version, nativeBuildInputs ? [], enableParallelBuilding ? true, ... }@args:
 
-let Dune = if args.useDune2 or false then dune_2 else dune_1; in
+# Force dune2 with OCaml >= 4.12
+let Dune = if args.useDune2 or (lib.versionAtLeast ocaml.version "4.12") then dune_2 else dune_1; in
```

Comment:
What’s the rationale for this behavior? You may want to change the default (for all versions of OCaml), but this is unrelated to the title of this PR.

Reply:
The point is that dune 1 isn't supported on ocaml >= 4.12. This change unblocks some packages that didn't build before on newer versions

Reply:
This looks error-prone to me: I’m going to forget to set `useDune2` because the package builds fine with default version of OCaml.

Reply:
Let me break out this commit to another PR so that we can have a discussion about it there. The other solution would be to add this to all packages I guess.